packages/github/README.md

```diff
@@ -55,3 +55,29 @@ const newIssue = await octokit.issues.create({
   body: 'Hello Universe!'
 });
 ```
+
+### Context Payload Casting in Typescript
+The [GitHub Webhook Event](https://developer.github.com/webhooks/#events) payload is provided as a part of the context. Based on the type of event that [triggered your workflow](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows), the structure of that object may change. You can cast this field to access these members in a type safe way.
+
+```ts
+const github = require('@actions/github');
+
+const payload = github.context.payload
```

Comment:
nit: `const payload = github.context.payload as Webhooks.WebhookPayloadPush` would eliminate the redundant casts further below

Reply:
I don't think we should recommend users do that though. Type assertions rely on the developer to perform relevant checks before casting, and this example shows that.

From their [docs](https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions)

Type assertions are a way to tell the compiler “trust me, I know what I’m doing.” A type assertion is like a type cast in other languages, but performs no special checking or restructuring of data. It has no runtime impact, and is used purely by the compiler. TypeScript assumes that you, ***the programmer, have performed any special checks that you need.***